### ISSUE TITLE
Priority of messages in Rebus

### ISSUE THREAD

**Issue by SorenA** [rebus-org/Rebus#645]
Hi!

Couldnt find anything in the documentation in relation to priorities, are they supported through Rebus?
We're using RabbitMQ so attatching it on a header level would be great.

**Comment by mookid8000** [comment]
Rebus does not readily support message priorities, as very few transports support it – however, it would definitely be possible to simply pass a specific header along with a message, and then have the RabbitMQ transport translate the header into a message priority somehow.

I have added it as an issue [here](https://github.com/rebus-org/Rebus.RabbitMq/issues/6)

**Comment by SorenA** [comment]
Great, thanks!

**Comment by mookid8000** [comment]
To people coming by this issue some time in the future: Rebus' RabbitMQ transport supports RabbitMQ's priority queues (available from RabbitMQ 3.5) via the `StrictPriorityqueue` configuration like this:

```csharp
Configure.With(activator)
    .Logging(l => l.Console(minLevel: LogLevel.Warn))
    .Transport(t => {
        t.UseRabbitMq("amqp://le-rabbit", "your-queue")
            .StrictPriorityQueue(10);
    })
    .Start();
```

and then by setting the priority of a message like this:

```csharp
var headers = new Dictionary<string, string>
{
    {RabbitMqHeaders.Priority, "5"}
};

await bus.Send(message, headers);
```

Please understand that the use of message priorities is subject to the limitations of RabbitMQ. E.g. it will fail if you attempt to treat an ordinary queue as a priority queue, or redefine the number of available priorities.
